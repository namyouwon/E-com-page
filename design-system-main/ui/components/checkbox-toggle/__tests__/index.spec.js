/* eslint-env jest */
import React from 'react';
import {
  CheckboxToggle,
  CheckboxToggleChecked,
  CheckboxToggleDisabled
} from '../base/example';

import createHelpers from '../../../../jest.helpers';

const { matchesMarkup } = createHelpers(__dirname);

it('renders a default checkbox toggle', () =>
  matchesMarkup(<CheckboxToggle />));

it('renders a checked checkbox toggle', () =>
  matchesMarkup(<CheckboxToggleChecked />));

it('renders a disabled checkbox toggle', () =>
  matchesMarkup(<CheckboxToggleDisabled />));
