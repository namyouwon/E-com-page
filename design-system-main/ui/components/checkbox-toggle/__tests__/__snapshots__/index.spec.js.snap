// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a checked checkbox toggle 1`] = `
<div
  className="slds-form-element"
>
  <label
    className="slds-checkbox_toggle slds-grid"
  >
    <span
      className="slds-form-element__label slds-m-bottom_none"
    >
      Toggle Label
    </span>
    <input
      aria-describedby="checkbox-toggle-2"
      defaultChecked={true}
      name="checkbox-toggle-2"
      type="checkbox"
      value="checkbox-toggle-2"
    />
    <span
      aria-live="assertive"
      className="slds-checkbox_faux_container"
      id="checkbox-toggle-2"
    >
      <span
        className="slds-checkbox_faux"
      />
      <span
        className="slds-checkbox_on"
      >
        Enabled
      </span>
      <span
        className="slds-checkbox_off"
      >
        Disabled
      </span>
    </span>
  </label>
</div>
`;

exports[`renders a default checkbox toggle 1`] = `
<div
  className="slds-form-element"
>
  <label
    className="slds-checkbox_toggle slds-grid"
  >
    <span
      className="slds-form-element__label slds-m-bottom_none"
    >
      Toggle Label
    </span>
    <input
      aria-describedby="checkbox-toggle-1"
      name="checkbox-toggle-1"
      type="checkbox"
      value="checkbox-toggle-1"
    />
    <span
      aria-live="assertive"
      className="slds-checkbox_faux_container"
      id="checkbox-toggle-1"
    >
      <span
        className="slds-checkbox_faux"
      />
      <span
        className="slds-checkbox_on"
      >
        Enabled
      </span>
      <span
        className="slds-checkbox_off"
      >
        Disabled
      </span>
    </span>
  </label>
</div>
`;

exports[`renders a disabled checkbox toggle 1`] = `
<div
  className="slds-form-element"
>
  <label
    className="slds-checkbox_toggle slds-grid"
  >
    <span
      className="slds-form-element__label slds-m-bottom_none"
    >
      Toggle Label
    </span>
    <input
      aria-describedby="checkbox-toggle-3"
      disabled={true}
      name="checkbox-toggle-3"
      type="checkbox"
      value="checkbox-toggle-3"
    />
    <span
      aria-live="assertive"
      className="slds-checkbox_faux_container"
      id="checkbox-toggle-3"
    >
      <span
        className="slds-checkbox_faux"
      />
      <span
        className="slds-checkbox_on"
      >
        Enabled
      </span>
      <span
        className="slds-checkbox_off"
      >
        Disabled
      </span>
    </span>
  </label>
</div>
`;
